### PR TITLE
feat: refactor bookmark search page

### PR DESCRIPTION
**Pull request by rebelchris**
In the new design we no longer have a separate search page for the bookmarks.
This refactor makes sure the search bar is always active for the bookmark page.

Screenshot 2021-12-14 at 17 15 27
Screenshot 2021-12-14 at 17 15 35

For the mobile view we added a new option to the main layout including `mobileTitle` this title will get rendered in the header for certain pages.
This does need the mobile sidebar to finalise how it renders.

### CODE REVIEW

**Review by sshanzel — COMMENTED** [review]
Just had a few very minor comments but very well done on working this relayout 💪

File: packages/shared/src/components/BookmarkFeedLayout.tsx [hunk 1]
```diff
@@ -55,21 +50,11 @@ export default function BookmarkFeedLayout({
   return (
     <FeedPage>
       {children}
+      <FeedPageHeader className="mb-5">
+        <h3 className="font-bold typo-callout">Bookmarks</h3>
+      </FeedPageHeader>
```
[the remark this answers is not in the document]
Comment (rebelchris) [reply packages/shared/src/components/BookmarkFeedLayout.tsx:55]: So far I haven't seen it used on any other pages like this.
The main layout uses another version of this.

File: packages/shared/src/components/utilities.tsx [hunk 2]
```diff
@@ -56,7 +56,12 @@ export const ActiveTabIndicator = classed(
 
 export const CustomFeedHeader = classed(
   'div',
-  'flex h-11 self-stretch items-center mb-6 text-theme-label-secondary typo-callout',
+  'flex h-11 self-stretch items-center mb-6 typo-callout',
+);
+
+export const FeedPageHeader = classed(
+  'nav',
```
[the remark this answers is not in the document]
Comment (rebelchris) [reply packages/shared/src/components/utilities.tsx:63]: Very good point!

File: packages/shared/src/components/BookmarkFeedLayout.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (sshanzel) [reply packages/shared/src/components/BookmarkFeedLayout.tsx:55]: Then I think this should be good to go 🚀

**Review by sshanzel — APPROVED** [review]
Woop! Woop! 🎉

File: packages/shared/src/components/utilities.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (idoshamun) [reply packages/shared/src/components/utilities.tsx]: Nice catch!

File: packages/shared/src/components/MainLayout.tsx [hunk 3]
```diff
@@ -89,6 +91,12 @@ export default function MainLayout({
           />
         )}
         <div className="flex-1" />
+        {mobileTitle && (
+          <strong className="block laptop:hidden typo-callout">
```
Comment (idoshamun) [inline packages/shared/src/components/MainLayout.tsx:95]: Shouldn't this be `hx` element?

**Review by idoshamun — CHANGES_REQUESTED** [review]
Looks great! Thanks for the info in the PR really helped me review it.
One UX question, what should happen when I hit the bookmarks button when I'm not logged in. Currently it shows a sign up popup. Is it the same now?

File: packages/shared/src/components/MainLayout.tsx (hunk 3 above)
Comment (rebelchris) [reply packages/shared/src/components/MainLayout.tsx:95]: I'm not sure what that means?

Comment (idoshamun) [reply packages/shared/src/components/MainLayout.tsx:95]: h1, h2, h3, h something

**Comment by sshanzel** [comment]
Was about to say the same for the reading history. Probably we should make it as a popup as well if we're gonna do that on the bookmarks to be consistent.

File: packages/shared/src/components/MainLayout.tsx (hunk 3 above)
Comment (rebelchris) [reply packages/shared/src/components/MainLayout.tsx:95]: We generally don't set the callout to a heading from what I can see.

Happy to change to a H3 as this is what it is on desktop, WDYT?
Do keep in mind this section will also render the good morning message (which is a span)
And the rank, (also a span)

**Comment by rebelchris** [comment]
@idoshamun / @sshanzel good call on that showLogin.
Actually missed that in the refactor.

Added it in this branch for reading history and bookmarks.
This takes priority over any other action that might have been set for the button.

File: packages/shared/src/components/MainLayout.tsx (hunk 3 above)
Comment (idoshamun) [reply packages/shared/src/components/MainLayout.tsx:95]: H3 sounds good

**Review by idoshamun — APPROVED** [review]
Actually you added the `showLogin` in the mobile branch which is perfectly fine by me

**Pull request merged**